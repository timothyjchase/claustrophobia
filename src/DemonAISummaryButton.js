import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { Button, Divider, Modal, Portal } from 'semantic-ui-react'

class DemonAISummaryButton extends PureComponent {
  constructor(props) {
    super(props)
    this.state = {
      open: props.open || false,
    }
  }

  handleOpen = () => {
    this.setState({ open: true })
  }

  handleClose = () => {
    this.setState({ open: false })
  }

  render() {
    const { as: ButtonComponent, ...buttonProps } = this.props
    const { open } = this.state

    // use a Portal in order to support a Dropdown.Item trigger component
    return (
      <Portal
        open={open}
        closeOnTriggerClickk={false}
        openOnTriggerClick
        trigger={<ButtonComponent {...buttonProps} />}
        onOpen={this.handleOpen}
        onClose={this.handleClose}
      >
        <Modal open closeOnDimmerClick={false}>
          <Modal.Header>Demon AI Summary</Modal.Header>
          <Modal.Content>
            <div>
              <p>
                <span style={{ color: 'red' }}>
                  <strong>Demon Die</strong>
                </span>
                : This die determines when a Demon will be added.
              </p>
              <p>
                During the Threat phase, if there is no demon in play, then a d6
                is rolled. If the result exceeds the Demon Die, then a demon is
                added and the Demon Die is reset to 6. If the result does not,
                then the Demon Die is reduced by 1.
              </p>
              <p>
                Where the Demon is added will depend on the scenario and the
                situation. The scenario will limit the number of times a Demon
                can be added.
              </p>
              <Divider />
              <p>
                <span style={{ color: 'blue' }}>
                  <strong>Theat Die</strong>
                </span>
                : This die primarily determines when Troglodytes will be added.
              </p>
              <p>
                During the Threat phase, if no demon has just been added, and
                the number of Troglodytes in play is less than the Threat Die,
                then Troglodytes will be added and then the Threat Die is
                reduced by 1.
              </p>
              <p>
                All Troglodytes may gain <strong>Supernatural Speed</strong>{' '}
                and/or <strong>Sharpened Claws</strong> depending on their
                proximity to Human warriors, further reducing the Threat Die.
                When the Threat Die reaches 0, it is rolled again.
              </p>
              <p>
                Where the Troglodytes are added will depend on the scenario and
                the situation.
              </p>
              <Divider />
              <p>
                <strong>Events</strong>: When a Troglogyte is removed, the Demon
                AI might draw an event card during the next Threat phase.
              </p>
              <p>
                The odds of drawing a card increase, the more Troglodytes are
                removed and the fewer that remain (1 / # of Troglodytes before
                removal). Removing a Troglodye, when there are 5 remaining,
                results in 20% chance of drawing an Event. Removing the last
                Troglodyte will guarantee that an event will be drawn.
              </p>
              <p>
                Only up to 1 Event will be drawn per turn. If the Event is not
                applicable, it will be discarded. For example, drawing{' '}
                <strong>Suicide Attack</strong>, when there are no Troglodytes
                in play, will be discarded.
              </p>
              <p>
                The <strong>Trap</strong> and <strong>They Are Legion</strong>{' '}
                actions from the Board of Destiny have been converted to Events.
              </p>
              <p />
            </div>
          </Modal.Content>
          <Modal.Actions>
            <Button content="Close" onClick={this.handleClose} />
          </Modal.Actions>
        </Modal>
      </Portal>
    )
  }
}

DemonAISummaryButton.defaultProps = {
  as: Button,
}

DemonAISummaryButton.propTypes = {
  as: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
}

export default DemonAISummaryButton
