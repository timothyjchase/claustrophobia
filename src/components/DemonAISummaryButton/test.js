import React from 'react'
import DemonAISummaryButton from './index'

describe('<DemonAISummaryButton />', () => {
  describe('with no props', () => {
    const component = shallow(<DemonAISummaryButton />)

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
