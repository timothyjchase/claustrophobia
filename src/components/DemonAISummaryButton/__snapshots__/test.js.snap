// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DemonAISummaryButton /> with no props should match snapshot 1`] = `
<Portal
  closeOnDocumentClick={true}
  closeOnEscape={true}
  closeOnTriggerClickk={false}
  eventPool="default"
  onClose={[Function]}
  onOpen={[Function]}
  open={false}
  openOnTriggerClick={true}
  trigger={
    <Button
      as="button"
      role="button"
    />
  }
>
  <Modal
    centered={true}
    closeOnDimmerClick={false}
    closeOnDocumentClick={false}
    dimmer={true}
    eventPool="Modal"
    open={true}
  >
    <ModalHeader>
      Demon AI Summary
    </ModalHeader>
    <ModalContent>
      <div>
        <Header
          as="h4"
        >
          <HeaderContent>
            <span
              style={
                Object {
                  "color": "red",
                }
              }
            >
              Demon Die
            </span>
            <HeaderSubheader>
              Determines when a Demon will be added
            </HeaderSubheader>
          </HeaderContent>
        </Header>
        <p>
          During the Threat phase, if there is no demon in play, then a d6 is rolled. If the result exceeds the Demon Die, then a demon is added and the Demon Die is reset to 6. If the result does not, then the Demon Die is reduced by 1.
        </p>
        <p>
          Where the Demon is added will depend on the scenario and the situation. The scenario will limit the number of times a Demon can be added.
        </p>
        <Divider />
        <Header
          as="h4"
        >
          <HeaderContent>
            <span
              style={
                Object {
                  "color": "blue",
                }
              }
            >
              Threat Die
            </span>
            <HeaderSubheader>
              Determines when Troglodytes will be added
              <br />
              (also Supernatural Speed or Sharpened Claws)
            </HeaderSubheader>
          </HeaderContent>
        </Header>
        <p>
          During the Threat phase, if no demon has just been added, and the number of Troglodytes in play is less than the Threat Die, then Troglodytes will be added and the Threat Die is reduced by 1.
        </p>
        <p>
          All Troglodytes may gain
          <strong>
             Supernatural Speed 
          </strong>
          and/or
          <strong>
             Sharpened Claws 
          </strong>
          depending on their proximity to Human warriors, further reducing the Threat Die (by 2 and 1 respectively). When the Threat Die reaches 0, it is rolled again.
        </p>
        <p>
          Where the Troglodytes are added will depend on the scenario and the situation.
        </p>
        <Divider />
        <Header
          as="h4"
        >
          <HeaderContent>
            Events
            <HeaderSubheader>
              Removing Troglogytes increases the chance that the Demon AI will draw an Event during the next Threat phase.
            </HeaderSubheader>
          </HeaderContent>
        </Header>
        <p>
          The odds of drawing a card increase, the more Troglodytes that are removed and the fewer that remain (percent of drawing an Event = 1 / total number of Troglodytes before removal). For example, removing 1 Troglodye, when there are 5 remaining, results in 20% chance of drawing an Event. Removing the last Troglodyte will guarantee that an event will be drawn.
        </p>
        <p>
          A maximum of 1 Event can be drawn per turn.
        </p>
        <p>
          The text of an Event may change depending on the scenario.
        </p>
        <p>
          The
          <strong>
             Trap 
          </strong>
          and
          <strong>
             They Are Legion 
          </strong>
          actions from the Board of Destiny have been converted into Events.
        </p>
        <p>
          Events may be cancelled with no effect, by using the upper right menu in the Event message (like when using
          <strong>
             Our Faith Will Protect Us
          </strong>
          ).
        </p>
      </div>
    </ModalContent>
    <ModalActions>
      <Button
        as="button"
        content="Close"
        onClick={[Function]}
        role="button"
      />
    </ModalActions>
  </Modal>
</Portal>
`;
